### PR TITLE
Update smtchecker tests to z3 4.8.13

### PR DESCRIPTION
**Pull request by leonardoalt**
Let's see how much nondeterminism we get...

### CODE REVIEW

File: test/libsolidity/smtCheckerTests/array_members/push_overflow_2_safe_no_overflow_assumption.sol [hunk 1]
```diff
@@ -13,4 +13,3 @@ contract C {
 // SMTEngine: all
 // SMTIgnoreOS: macos
 // ----
-// Info 1180: Contract invariant(s) for :C:\n(x.length >= 0)\n
```
Comment (cameel) [inline test/libsolidity/smtCheckerTests/array_members/push_overflow_2_safe_no_overflow_assumption.sol:16]: Is that non-determinism too?

File: test/libsolidity/smtCheckerTests/types/address_staticcall.sol [hunk 2]
```diff
@@ -10,14 +10,16 @@ contract C
 		assert(success);
 		assert(x == 0);
 		assert(map[0] == 0);
-		assert(localMap[0] == 0);
+		// Disabled because of Spacer's seg fault
+		//assert(localMap[0] == 0);
```
Comment (cameel) [inline test/libsolidity/smtCheckerTests/types/address_staticcall.sol:14]: Is that on macOS again or on Linux too?

File: test/libsolidity/smtCheckerTests/array_members/push_overflow_2_safe_no_overflow_assumption.sol (hunk 1 above)
Comment (leonardoalt) [reply test/libsolidity/smtCheckerTests/array_members/push_overflow_2_safe_no_overflow_assumption.sol:16]: If it happens on all OSs it should be fine.

File: test/libsolidity/smtCheckerTests/types/address_staticcall.sol (hunk 2 above)
Comment (leonardoalt) [reply test/libsolidity/smtCheckerTests/types/address_staticcall.sol:14]: Also on Linux, the current state of this PR is what I got locally.

**Comment by leonardoalt** [comment]
I think I resolved the current nondeterminism issues.
There are still 3 things to do on this PR:
- [x] Fix the z3 hash for the OSX build
- [x] Fix the intx thing so that the last docker build in https://github.com/ethereum/solidity/pull/12312 goes through, and use that final hash to update the 16.04 image used by CI here
- [x] Fix the remaining nondeterminism problems from the OSX build

File: test/libsolidity/smtCheckerTests/types/address_staticcall.sol (hunk 2 above)
Comment (cameel) [reply test/libsolidity/smtCheckerTests/types/address_staticcall.sol:14]: Is that maybe this? https://github.com/Z3Prover/z3/issues/4690

Comment (leonardoalt) [reply test/libsolidity/smtCheckerTests/types/address_staticcall.sol:14]: Yea I guess

**Comment by leonardoalt** [comment]
Ok I think this can be merged

**Review by chriseth — APPROVED** [review]

**Pull request merged**